pkg/registry-disk/registry-disk_test.go

```diff
@@ -0,0 +1,196 @@
+ *
+ * Copyright 2017 Red Hat, Inc.
+ *
+ */
+
+package registrydisk
+
+import (
+	"io/ioutil"
+	"os"
+	"os/user"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	v1 "kubevirt.io/kubevirt/pkg/api/v1"
+	diskutils "kubevirt.io/kubevirt/pkg/ephemeral-disk-utils"
+)
+
+var _ = Describe("RegistryDisk", func() {
+	tmpDir, _ := ioutil.TempDir("", "registrydisktest")
+	owner, err := user.Current()
+	if err != nil {
+		panic(err)
+	}
+
+	VerifyDiskType := func(diskExtension string) {
+		vm := v1.NewMinimalVM("fake-vm")
+		vm.Spec.Domain.Devices.Disks = append(vm.Spec.Domain.Devices.Disks, v1.Disk{
+			Type:   "RegistryDisk:v1alpha",
+			Device: "disk",
+			Source: v1.DiskSource{
+				Name: "someimage:v1.2.3.4",
+			},
+			Target: v1.DiskTarget{
+				Device: "vda",
+			},
+		})
+
+		// create a fake disk file
+		volumeMountDir := generateVMBaseDir(vm)
+		err = os.MkdirAll(volumeMountDir+"/disk0", 0750)
+		Expect(err).ToNot(HaveOccurred())
+
+		filePath := volumeMountDir + "/disk0/disk-image." + diskExtension
+		_, err := os.Create(filePath)
+
+		vm, err = MapRegistryDisks(vm)
+		Expect(err).ToNot(HaveOccurred())
+
+		// verify file gets renamed by virt-handler to prevent container from
+		// removing it before VM is exited
+		exists, err := diskutils.FileExists(filePath)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(exists).To(Equal(false))
+
+		// verify file rename takes place
+		exists, err = diskutils.FileExists(filePath + ".virt")
+		Expect(err).ToNot(HaveOccurred())
+		Expect(exists).To(Equal(true))
+
+		Expect(vm.Spec.Domain.Devices.Disks[0].Type).To(Equal("file"))
+		Expect(vm.Spec.Domain.Devices.Disks[0].Target.Device).To(Equal("vda"))
+		Expect(vm.Spec.Domain.Devices.Disks[0].Driver).ToNot(Equal(nil))
+		Expect(vm.Spec.Domain.Devices.Disks[0].Driver.Type).To(Equal(diskExtension))
+		Expect(vm.Spec.Domain.Devices.Disks[0].Source).ToNot(Equal(nil))
+		Expect(vm.Spec.Domain.Devices.Disks[0].Source.File).To(Equal(filePath + ".virt"))
+
+		err = CleanupEphemeralDisks(vm)
+		exists, err = diskutils.FileExists(volumeMountDir)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(exists).To(Equal(false))
+	}
+
+	BeforeSuite(func() {
+		err := SetLocalDirectory(tmpDir)
+		if err != nil {
+			panic(err)
+		}
+		SetLocalDataOwner(owner.Username)
+	})
+
+	AfterSuite(func() {
+		os.RemoveAll(tmpDir)
+	})
+
+	Describe("registry-disk", func() {
+		Context("verify helper functions", func() {
+			It("by verifying error when no disk is present", func() {
+
+				vm := v1.NewMinimalVM("fake-vm")
+				vm.Spec.Domain.Devices.Disks = append(vm.Spec.Domain.Devices.Disks, v1.Disk{
+					Type:   "RegistryDisk:v1alpha",
+					Device: "disk",
+					Source: v1.DiskSource{
+						Name: "someimage:v1.2.3.4",
+					},
+					Target: v1.DiskTarget{
+						Device: "vda",
+					},
+				})
+
+				vm, err := MapRegistryDisks(vm)
+				Expect(err).To(HaveOccurred())
+			})
+
+			It("by verifying mapping of qcow2 disk", func() {
+				VerifyDiskType("qcow2")
```

Comment:
You can also use a table to desciribe these two "It"


```
table.DescribeTable("whatever", func(type string) {

}, 
table.Entry("whatever ", "qcow2"),
table.Entry("whatever ", "raw"),
)

```

Reply:
good idea